src/test/test_path_cost.py

```diff
@@ -0,0 +1,82 @@
+import unittest
+from unittest.mock import MagicMock
+from ChessEnums import Player, Piece_Type
+import State
+from PathCost import PathCost
+
+class TestPathCost(unittest.TestCase):
```

Comment:
TODO:
Investigate warning:

`DeprecationWarning: PathCost.__float__ returned non-float (type numpy.float64).  The ability to return an instance of a strict subclass of float is deprecated, and may be removed in a future version of Python.
    self.assertAlmostEqual(float(path_cost), expected_piece_weight, delta=0.01)`